python/lsst/ts/envsensors/device_config.py

```diff
@@ -59,14 +65,18 @@ def as_dict(self) -> Dict[str, Union[str, int]]:
 
         Returns
         -------
-        dict: `dict`
+        device_config_as_dict: `dict`
             A dictionary of key-value pairs representing the instance
             attributes and their values.
         """
-        return {
+        device_config_as_dict: Dict[str, Union[str, int]] = {
             Key.NAME: self.name,
-            Key.CHANNELS: self.num_channels,
             Key.DEVICE_TYPE: self.dev_type,
             Key.FTDI_ID: self.dev_id,
             Key.SENSOR_TYPE: self.sens_type,
         }
+
+        # Only temperature sensors have a configurable number of channels.
+        if self.sens_type == SensorType.TEMPERATURE:
+            device_config_as_dict[Key.CHANNELS] = self.num_channels
```

Comment:
This is a confusing to me because these new HX85 sensors also measure temperature. I encourage you to find a less generic name for the `TemperatureSensor` class.

Reply:
In the ESS context, the sensors are always referred to by their principal function. The temperature sensors measure temperatures only and are referred top as such. Despite the fact that the humidity sensors measure more, they are still referred to as humidity sensors. For the ESS people this is 100% clear. I therefore prefer to keep the naming as is.

Reply:
Yesterday we discovered that there in fact are two kinds of temperature sensors (apart from the humidity sensors that also measure temperature). Once the differences between the two have been described, I will change the code to reflect that and will introduce two temperature sensor classes with less generic names.